### PR TITLE
Add `Actor` to `add_mesh` return type

### LINKED ISSUE #6257: Return type hint in plotting/plotter.py
### Describe the feature you would like to be added.

Hello, I would like to have a return type hint for methods in `plotting/plotter.py`.

For example with `add_mesh()` I would like to have a return type `Actor` so that my IDE (VS code) understand that the variable is an `Actor` and make it easier to navigate into the code.

In short, I would like to go from the first picture to the second:
image
image

I wanted to make a PR but it seems I need to fix some additional ruff rules and mypy errors shown by pre-commit so I am a bit stuck. 
Is there a way to implement the return type hints? Or maybe a different way to solve my problem?

Thanks

### Links to VTK Documentation, Examples, or Class Definitions.

_No response_

### Pseudocode or Screenshots

_No response_

### PR DESCRIPTION
**Pull request by MatthewFlamm**
### Overview

closes #6257

### CODE REVIEW

File: pyvista/plotting/plotter.py [hunk 1]
```diff
@@ -3728,7 +3728,7 @@ def add_mesh(
         prop = Property(**prop_kwargs)
         actor.SetProperty(prop)
 
-        if style == 'points_gaussian':
+        if style == 'points_gaussian' and self.mapper.dataset is not None:
```
Comment (MatthewFlamm) [inline pyvista/plotting/plotter.py:3731]: `self.mapper.dataset` type is `DataSet | None`.  This is probably the right fix from a typing standpoint.  It is _possible_ that it should be only `DataSet` here and a different fix is needed, but given the complexity of `add_mesh` I'm not confident that it is true.

**Comment by MatthewFlamm** [comment]
@kmarchais

File: pyvista/plotting/plotter.py (hunk 1 above)
Comment (MatthewFlamm) [reply pyvista/plotting/plotter.py:3731]: If it is `None`, then we just stop setting the parameters for `point_gaussian` which seems like an okay result to me.

**Comment by kmarchais** [comment]
Thank you ! 

Maybe we can take this opportunity to add more return type hints on this file in this PR?

**Comment by kmarchais** [comment]
On this branch on my fork [kmarchais/pyvista/pyvista/plotting/plotter.py](https://github.com/kmarchais/pyvista/blob/maint/add-mesh-return-type/pyvista/plotting/plotter.py), I have added return types for almost all other functions except for `show` if you want to use it.

**Comment by tkoyama010** [comment]
> On this branch on my fork [kmarchais/pyvista/pyvista/plotting/plotter.py](https://github.com/kmarchais/pyvista/blob/maint/add-mesh-return-type/pyvista/plotting/plotter.py), I have added return types for almost all other functions except for `show` if you want to use it.

@kmarchais 
Sorry to interrupt, but you can also create a Pull Request for `maint/add-mesh-return-type`. That would work better as your contribution would be clearer in my mind.

**Comment by kmarchais** [comment]
Oh right! Thanks, I have just done it

File: pyvista/plotting/plotter.py (hunk 1 above)
Comment (adeak) [reply pyvista/plotting/plotter.py:3736]: Yup, I'm not at all aware of all the mechanics here, but the fix seems to err on the safe side.

**Review by adeak — APPROVED** [review]
LGTM, thanks.

**Review by tkoyama010 — APPROVED** [review]
LGTM

**Pull request merged**